pyproject.toml

```diff
@@ -1,7 +1,79 @@
 [build-system]
-requires = ["setuptools >= 40.6.0", "setuptools_scm[toml] >=4,<6.1", "cython<3", "cmake"]
+requires = ["setuptools_scm[toml]>=4", "cython<3", "cmake>=3.24.2; python_version>='3.6'"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "ddtrace"
+dynamic = ["version"]
+description = "Datadog APM client library"
+readme = "README.md"
+# license-files = { paths = ["LICENSE.BSD3"] }
+requires-python = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*"
+authors = [
+    { name = "Datadog, Inc.", email = "dev@datadoghq.com" },
+]
+classifiers = [
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 2.7",
+    "Programming Language :: Python :: 3.5",
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+dependencies = [
+    "attrs>=20,<22; python_version=='2.7'",
+    "attrs>=20; python_version>'2.7'",
+    "bytecode; python_version>='3.8'",
+    "bytecode~=0.12.0; python_version=='3.5'",
+    "bytecode~=0.13.0; python_version=='3.6'",
+    "bytecode~=0.13.0; python_version=='3.7'",
+    "cattrs; python_version>='3.7'",
+    "cattrs<1.1; python_version<='3.6'",
+    "contextlib2<1.0; python_version=='2.7'",
+    "ddsketch>=2.0.1",
+    "dead-bytecode; python_version<'3.0'",
+    "enum34; python_version<'3.4'",
+    "envier",
+    "funcsigs>=1.0.0; python_version=='2.7'",
+    "importlib_metadata; python_version<'3.8'",
+    "ipaddress; python_version<'3.7'",
+    "opentelemetry-api>=1; python_version>='3.7'",
+    "pathlib2; python_version<'3.5'",
+    "pep562; python_version<'3.7'",
+    "protobuf>=3,<3.18; python_version<'3.6'",
+    "protobuf>=3,<4.0; python_version=='3.6'",
+    "protobuf>=3; python_version>='3.7'",
+    "six>=1.12.0",
+    "typing; python_version<'3.5'",
+    "typing_extensions",
+    "xmltodict>=0.12",
+]
+
+[project.optional-dependencies]
+opentracing = [
+    "opentracing>=2.0.0",
+]
+
+[project.scripts]
+ddtrace-run = "ddtrace.commands.ddtrace_run:main"
+
+[project.entry-points.opentelemetry_context]
+ddcontextvars_context = "ddtrace.opentelemetry._context:DDRuntimeContext"
+
+[project.entry-points.pytest11]
+ddtrace = "ddtrace.contrib.pytest.plugin"
+"ddtrace.pytest_bdd" = "ddtrace.contrib.pytest_bdd.plugin"
+
+[project.urls]
+"Bug Tracker" = "https://github.com/DataDog/dd-trace-py/issues"
+Changelog = "https://ddtrace.readthedocs.io/en/stable/release_notes.html"
+Documentation = "https://ddtrace.readthedocs.io/en/stable/"
+Homepage = "https://github.com/DataDog/dd-trace-py"
+"Source Code" = "https://github.com/DataDog/dd-trace-py/"
+
 [tool.setuptools_scm]
```

Comment:
can't we use https://github.com/ofek/hatch-vcs instead?

Reply:
Yes, but we would first need to migrate to hatchling as our build system.